### DISCUSSION TITLE
What do the 'Specific Time' functions in RemoteControl api/stelaction do?

### DISCUSSION THREAD

**Discussion by DrRichardDaystrom** [Stellarium/stellarium#3925] [Q&A] [answered]
I've been exploring the RemoteControl API and I see that there are a whole set of 'Specific Time' functions available as stelactions - e.g. `actionPrevious_March_Equinox`, `actionCurrent_June_Solstice`, `actionNext_December_Solstice`, etc.

What do these functions do?  When I call them with POST `api/stelaction/do`, and pass the stelaction id, it doesn't appear to do anything and nothing is returned in the http response.

I presumed from the names that these stelactions would set the time to the corresponding events but that doesn't seem to happen.  What is supposed to happen when these stelactions are called?

**Comment by alex-w** [comment]
Go to the specific moments of the time :) 

For example calling action `actionCurrent_June_Solstice` will set the time of summer solstice (northern hemisphere) in current year.

**Reply by DrRichardDaystrom** [reply]
So it does - but only back until the year -1000.  When I try to give a date earlier than that, it doesn't seem to do anything.

I discovered the issue working on a script for the year -1800 so I initially thought it didn't work at all.  When you said it worked, I checked with more recent dates and saw that you were right.  It was pretty quick to track down that it breaks before -1000.

Is this an ephemeris issue?  Can I fix it by loading a different ephemeris?

Sample code demonstrating the issue, in Python ...

```
import requests
import juliandate as jd

# specify URL to Stellarium API
url_stellarium = 'http://localhost:8090/api/'

# specify observation date as a Julian day (jd) starting at midday
obs_jd = jd.from_gregorian(-1001, 3, 21) + 0.5

# set Stellarium time to obs_jd & timerate to zero (stopped)
r_time = requests.post(url_stellarium + 'main/time', data = {'time': str(obs_jd), 'timerate': '0'})

# POST stelacation/do
r_do = requests.post(url_stellarium + 'stelaction/do', data = {'id': 'actionCurrent_June_Solstice'})
```

**Reply by alex-w** [reply]
Oops... this is effect for application of the error code which is used as "invalid" value of time.

P.S. I didn't checked the code, but value -1000 is very specific for RTS time feature

**Reply by alex-w** [reply]
I was wrong. The algorithm for calculation of solstices and equinoxes has valid range of application between years -1000 and 3000. So, this is not a bug, this is a feature.

**Reply by DrRichardDaystrom** [reply]
Interesting. But the _VSOP87_ planetary theory is good back to -4000 … right?  So if I track the Sun to find the solstices / equinoxes, the results should be accurate - even before -1000?

**Reply by alex-w** [reply, marked as answer]
> Interesting. But the _VSOP87_ planetary theory is good back to -4000 … right? So if I track the Sun to find the solstices / equinoxes, the results should be accurate - even before -1000?

We use Meeus' quick algorithms for finding the time of equinoxes and solstices. Of course you can show equator and ecliptic and finding extremums by hands via VSOP87/DE4xx.

**Reply by DrRichardDaystrom** [reply]
I’ll give that a try. Thanks.
